_ml-commons-plugin/api/model-apis/register-model.md

```diff
@@ -0,0 +1,128 @@
+
+You can further customize a pretrained sentence transformer model's post-processing logic with the following optional fields in the `model_config` object.
+
+| Field | Data type | Description |
+| :--- | :--- | :--- |
+| `pooling_mode` | String | The post-process model output, either `mean`, `mean_sqrt_len`, `max`, `weightedmean`, or `cls`.|
+| `normalize_result` | Boolean | When set to `true`, normalizes the model output in order to scale to a standard range for the model. |
+
+#### Example request
+
+The following example request registers a version `1.0.0` of an NLP sentence transformation model named `all-MiniLM-L6-v2`.
+
+```json
+POST /_plugins/_ml/models/_register
+{
+    "name": "all-MiniLM-L6-v2",
+    "version": "1.0.0",
+    "description": "test model",
+    "model_format": "TORCH_SCRIPT",
+    "model_group_id": "FTNlQ4gBYW0Qyy5ZoxfR",
+    "model_content_hash_value": "c15f0d2e62d872be5b5bc6c84d2e0f4921541e29fefbef51d59cc10a8ae30e0f",
+    "model_config": {
+        "model_type": "bert",
+        "embedding_dimension": 384,
+        "framework_type": "sentence_transformers",
+       "all_config": "{\"_name_or_path\":\"nreimers/MiniLM-L6-H384-uncased\",\"architectures\":[\"BertModel\"],\"attention_probs_dropout_prob\":0.1,\"gradient_checkpointing\":false,\"hidden_act\":\"gelu\",\"hidden_dropout_prob\":0.1,\"hidden_size\":384,\"initializer_range\":0.02,\"intermediate_size\":1536,\"layer_norm_eps\":1e-12,\"max_position_embeddings\":512,\"model_type\":\"bert\",\"num_attention_heads\":12,\"num_hidden_layers\":6,\"pad_token_id\":0,\"position_embedding_type\":\"absolute\",\"transformers_version\":\"4.8.2\",\"type_vocab_size\":2,\"use_cache\":true,\"vocab_size\":30522}"
+    },
+    "url": "https://artifacts.opensearch.org/models/ml-models/huggingface/sentence-transformers/all-MiniLM-L6-v2/1.0.1/torch_script/sentence-transformers_all-MiniLM-L6-v2-1.0.1-torch_script.zip"
+}
+```
+{% include copy-curl.html %}
+
+#### Example response
+
+OpenSearch responds with the `task_id` and task `status`.
+
+```json
+{
+  "task_id" : "ew8I44MBhyWuIwnfvDIH", 
+  "status" : "CREATED"
+}
+```
+
+## Check the status of model registration
+
+To see the status of your model registration and retrieve the model ID created for the new model version, pass the `task_id` as a path parameter to the Tasks API:
+
+```json
+GET /_plugins/_ml/tasks/<task_id>
+```
+{% include copy-curl.html %}
+
+The response contains the model ID of the model version:
+
+```json
+{
+  "model_id": "Qr1YbogBYOqeeqR7sI9L",
+  "task_type": "DEPLOY_MODEL",
+  "function_name": "TEXT_EMBEDDING",
+  "state": "COMPLETED",
+  "worker_node": [
+    "N77RInqjTSq_UaLh1k0BUg"
+  ],
+  "create_time": 1685478486057,
+  "last_update_time": 1685478491090,
+  "is_async": true
+}
+```
+
+## Registering a model containing an internal connector
```

Comment:
Refer to https://github.com/opensearch-project/ml-commons/blob/2.x/docs/tutorials/remote_inference.md

For Register remote model, user can use internal connector or connector id (standalone connector)